File: server/php/src/Payment/PaymentDetailsService.php
<?php

namespace MyApp\Payment;
use OnlinePayments\Sdk\Domain\PaymentDetailsResponse;
use OnlinePayments\Sdk\Merchant\MerchantClientInterface;

/**
 * Payment details service
 */
class PaymentDetailsService
{
    private $merchantClient;

    public function __construct(MerchantClientInterface $merchantClient) {
        $this->merchantClient = $merchantClient;
    }

    public function getPaymentDetails(string $paymentId): PaymentDetailsResponse {
        return $this->merchantClient->payments()->getPaymentDetails($paymentId);
    }

    public function getPaymentDetailsForHostedCheckout(string $hostedCheckoutId): PaymentDetailsResponse {
        $paymentId = sprintf("%s_0", $hostedCheckoutId);
        return $this->merchantClient->payments()->getPaymentDetails($paymentId);
    }
}

File: server/java/src/main/java/com/worldlinesolutions/onlinepayments/example/payment/PaymentDetailsService.java
package com.worldlinesolutions.onlinepayments.example.payment;

import com.onlinepayments.domain.PaymentDetailsResponse;
import com.onlinepayments.merchant.MerchantClientInterface;
import org.springframework.stereotype.Service;

/**
 * Payment details service used for fetching payment details
 */
@Service
public class PaymentDetailsService {

    private final MerchantClientInterface merchantClient;

    public PaymentDetailsService(
            MerchantClientInterface merchantClientFromCommunicator
    ) {
        this.merchantClient = merchantClientFromCommunicator;
    }

    /**
     * Gets payment details for a hostedCheckoutId
     * @param hostedCheckoutId Id of the hosted checkout
     * @return Instance of {@link PaymentDetailsResponse}
     */
    public PaymentDetailsResponse getPaymentDetailsForHostedCheckout(String hostedCheckoutId) {
        String paymentId = String.format("%s_0", hostedCheckoutId);
        return merchantClient.payments().getPaymentDetails(paymentId);
    }

    /**
     * Gets payment details for a payment id
     * @param paymentId Id of the payment
     * @return Instance of {@link PaymentDetailsResponse}
     */
    public PaymentDetailsResponse getPaymentDetails(String paymentId) {
        return merchantClient.payments().getPaymentDetails(paymentId);
    }
}
